tree/ntuple/v7/inc/ROOT/RColumnModel.hxx

```diff
@@ -43,9 +43,9 @@ When changed, remember to update
 enum class EColumnType {
    kUnknown = 0,
    // type for root columns of (nested) collections; 32bit integers that count relative to the current cluster
-   kIndex,
-   // 64 bit column that uses the lower 32bits as kIndex and the higher 32bits as a dispatch tag; used, e.g.,
-   // in order to serialize std::variant
+   kIndex32,
+   // 64 bit column that uses the lower 44 bits like kIndex64, higher 20 bits are a dispatch tag to a column ID;
```

Comment:
Will `kIndex64` be define in a subsequent PR or is it from another enum?

Reply:
A confusing part is `kIndex32 ... that uses the lower 44 bits like` (plus On-disk representation stays as is, 32bit.)  What happens to the 12 extra bits and/or what am I missing?

Reply:
Yep, I split this in two PRs.  The next PR will have `kIndex64`, `kSplitIndex32`, and `kSplitIndex64`

Reply:
The 44bit/20bit split is for the `kSwitch` column, which can at the same time select a column and address an element in that column.  The index columns also address an element in a different column, but the target column is fixed.

Reply:
I see (in part because of the way the diff look) I apply the comment to the previous rather than the next line ... 
The pattern:
```
  kA,
  // some comment (implicitly about kB)
  kB,
  kC,
  // some comment (implicitly about kD)
  kD,
```
is confusing to me (even-though it does follow the same pattern we use elsewhere) because of the sparseness  which makes me (maybe it just me) read it by implicitly suppressing the new line:
```
  kA,  // some comment (implicitly about kB)
  kB,
  kC,  // some comment (implicitly about kD)
  kD,
```